plugin/backend_linux.go

```diff
@@ -52,12 +60,31 @@ func (pm *Manager) Enable(name string, config *types.PluginEnableConfig) error {
 }
 
 // Inspect examines a plugin config
-func (pm *Manager) Inspect(name string) (tp types.Plugin, err error) {
-	p, err := pm.pluginStore.GetByName(name)
-	if err != nil {
+func (pm *Manager) Inspect(refOrID string) (tp types.Plugin, err error) {
+	if pluginName, err := getPluginName(refOrID); err == nil {
+		if p, err := pm.pluginStore.GetByName(pluginName); err == nil {
+			return p.PluginObj, nil
+		}
+	}
+
+	pluginID := strings.ToLower(refOrID)
```

Comment:
Is the order correct here? I think we should try;

1. Match on full ID
2. Match on full name
3. Match on partial ID

Reason for this is so that a name (which is user-configurable) should never be able to mask ("spoof") an ID, and a partial ID (`db`) should not mask the full name of the container